go/scheduler/tasks.py

```diff
@@ -0,0 +1,65 @@
+import datetime
+
+from go.vumitools.tests.helpers import djangotest_imports
+
+with djangotest_imports(globals()):
+    from celery.task import task, group
+
+    from go.scheduler.models import PendingTask, Task
+    from go.base.utils import (
+        get_conversation_view_definition, vumi_api)
+
+
+@task()
+def perform_task(pending_id):
+    """ Perform a task. """
+    pending = PendingTask.objects.get(id=pending_id)
+    if pending.started_timestamp is None:
+        pending.started_timestamp = datetime.datetime.utcnow()
+        pending.save()
+    else:
+        return
```

Comment:
I'm not sure here if we should delete the task or not.

If we don't delete it, then we could get into the situation where every time we try to run the task, we load it from the database and then just return.

If we do delete it, then we could run into the trace condition of getting an error because we delete it before the original task runner finishes, and then the original task runner tries to delete it, but it doesn't exist.